src/enzo/ActiveParticle.h

```diff
@@ -462,6 +462,68 @@ namespace ActiveParticleHelpers {
 
   }
 
+  template <class APClass> std::vector<hid_t> GetParticleAttributesHDF5DataType() {
+      std::vector<hid_t> attribute_datatype;
+
+      AttributeVector &handlers = APClass::AttributeHandlers;
+      for (AttributeVector::iterator it = handlers.begin(); it != handlers.end(); ++it) {
+          attribute_datatype.push_back((*it)->hdf5type);
+      }
+
+      return attribute_datatype;
+  }
+
+  template <class APClass> std::vector<void*> GetParticleAttributes(ActiveParticleList<ActiveParticleType>& InList,
+          int ParticleTypeID, int TotalParticles, int Count, const std::string& particle_name) {
+
+      // This function allocates and returns the new buffer for particle attributes,
+      // and it is the caller's responsibility to free the buffer when it doesn't need it at all.
+      // It also ignores multi-dimensional arrays.
+      // Currently, this function is solely used in libyt.
+
+      std::vector<void*> attribute_values;
+
+      AttributeVector &handlers = APClass::AttributeHandlers;
+      for (AttributeVector::iterator it = handlers.begin(); it != handlers.end(); ++it) {
+          const char *attr_name = (*it)->name.c_str();
+
+          // ignores multi-dimensional arrays
+          if ((strcmp(attr_name, "AccretionRate") == 0 || strcmp(attr_name, "AccretionRateTime") == 0 || strcmp(attr_name, "Accreted_angmom") == 0)
```

Comment:
Instead of enumerating all the known vector fields, we can check if it's a 1D vector by comparing the `element_size` attribute to the result of `H5Tget_size(hdf5type)`.  If the two are equal, it is not 2D.

Reply:
`element_size` is set to `N*sizeof(Type)`

Reply:
@matthewturk 
I had a second thoughts on this. 
It looks like `element_size` returns `sizeof(Type) * N` only in `ArrayHandler` class when the attribute doesn't start with `particle_`. ([`ArrayHandler`](https://github.com/data-exp-lab/enzo-dev/blob/main/src/enzo/ParticleAttributeHandler.h#L142-L145) and [`Handler`](https://github.com/data-exp-lab/enzo-dev/blob/main/src/enzo/ParticleAttributeHandler.h#L82)).
Which means we cannot use it to catch the dimension of it.

Reply:
Oh, I think we can! All of them are `ArrayHandler` class. Please ignore the last comments. :sweat: